### DISCUSSION TITLE
vcpkg list --host-triplet="x64-windows" also list for "x64-windows-cpp17"

### DISCUSSION THREAD

**Discussion by playgithub** [microsoft/vcpkg#23997] [Ideas]
vcpkg list --host-triplet="x64-windows" also list for "x64-windows-cpp17"
Why not list for "x64-windows" only?

```cmd
> vcpkg list --host-triplet="x64-windows"
...
sqlite3:x64-windows                                3.37.2#1         SQLite is a software library that implements a s... sqlite3:x64-windows-cpp17                          3.37.2#1         SQLite is a software library that implements a s... tiff:x64-windows-cpp17                             4.3.0#6          A library that supports the manipulation of TIFF... tiff[jpeg]:x64-windows-cpp17                                        Support JPEG compression in TIFF image files
tiff[lzma]:x64-windows-cpp17                                        Support LZMA compression in TIFF image files
tiff[zip]:x64-windows-cpp17                                         Support ZIP/deflate compression in TIFF image files vcpkg-cmake-config:x64-windows                     2022-02-06
vcpkg-cmake:x64-windows                            2022-01-19
zlib:x64-windows                                   1.2.11#13        A compression library
zlib:x64-windows-cpp17                             1.2.11#13        A compression library
zstd:x64-windows-cpp17                             1.5.2#1          Zstandard - Fast real-time compression algorithm
...
```

**Comment by Neumann-A** [comment]
`--host-triplet=` does nothing here

you want 
`vcpkg list x64-windows`
the argument to the list command does the filtering.

**Comment by playgithub** [comment]
> `--host-triplet=` does nothing here
> 
> you want `vcpkg list x64-windows` the argument to the list command does the filtering.

`vcpkg list x64-windows` lists for both `x64-windows` and `x64-windows-cpp17`

However, `vcpkg list x64-windows-cpp17` lists for  only `x64-windows-cpp17`

**Comment by Cheney-W** [comment]
The parameter of the list command will be treated as a string to match the content in the ${VCPKG_INSTALLED_DIR}/vcpkg/info folder, then parse the corresponding list file whose file name contains this string into following format for output.
port name      version     discription

According to the above rules, because "x64-windows-cpp17" contains "x64-windows", the result of two triplets is found.

**Comment by playgithub** [comment]
> The parameter of the list command will be treated as a string to match the content in the ${VCPKG_INSTALLED_DIR}/vcpkg/info folder, then parse the corresponding list file whose file name contains this string into following format for output. port name version discription
> 
> According to the above rules, because "x64-windows-cpp17" contains "x64-windows", the result of two triplets is found.

Hope for an improvment for this.
